src/gfx/reverse.cpp

```diff
@@ -241,24 +242,27 @@ void reverse() {
 
 	if (tilemap) {
 		if (attrmap) {
-			for (auto [id, attr] : zip(*tilemap, *attrmap)) {
-				bool bank = attr & 1 << 3;
-				if (id >= options.maxNbTiles[bank]) {
+			for (auto [tileID, attribute] : zip(*tilemap, *attrmap)) {
+				if (bool bank = attribute & 1 << 3; tileID >= options.maxNbTiles[bank]) {
 					warning(
-					    "Tile #%" PRIu8 " was referenced, but the limit for bank %u is %" PRIu16,
-					    id,
+					    "Tilemap references tile #%" PRIu8 ", but the limit for bank %u is %" PRIu16,
+					    tileID,
 					    bank,
 					    options.maxNbTiles[bank]
 					);
 				}
 			}
 		} else {
-			for (auto id : *tilemap) {
-				if (id >= options.maxNbTiles[0]) {
-					warning(
-					    "Tile #%" PRIu8 " was referenced, but the limit is %" PRIu16,
-					    id,
-					    options.maxNbTiles[0]
+			size_t limit = std::min<size_t>(nbTiles, options.maxNbTiles[0]) + options.baseTileIDs[0];
```

Comment:
Not quite! It's possible for the “allowed” range to wrap around (e.g. `-N 128 -b 192`), so we want to compare `static_cast<uint8_t>(tileID - options.baseTileIDs[bank])`[^int] against `limit(bank)`[^lim].

Though, then the error message can become confusing; I'd suggest turning it into `...at ({tx}, {ty}), outside of the allowed [{options.baseTileIDs[bank]}, {static_cast<uint8_t>(options.baseTileIDs[bank] + limit(bank))}]`.

[^int]: Yes, that cast is necessary. Fuck integer promotion.
[^lim]: I don't remember the behaviour if tiles are referenced from bank 1, but `-N` is not passed to give its size?

Reply:
I'm not sure how or if to account for `baseTileIDs` here (see the other issue comment).